File: jdroid-android-core/src/main/java/com/jdroid/android/androidx/viewmodel/AbstractViewModel.java
package com.jdroid.android.androidx.viewmodel;

import com.jdroid.android.utils.TagUtils;
import com.jdroid.java.utils.LoggerUtils;

import androidx.annotation.CallSuper;
import androidx.lifecycle.ViewModel;

public abstract class AbstractViewModel extends ViewModel {

	@CallSuper
	@Override
	protected void onCleared() {
		super.onCleared();
		LoggerUtils.getLogger(getTag()).info("Clearing view model");
	}

	protected String getTag() {
		return TagUtils.getTag(getClass());
	}
}

File: jdroid-android-core/src/main/java/com/jdroid/android/androidx/viewmodel/AbstractViewModel.kt
package com.jdroid.android.androidx.viewmodel

import androidx.annotation.CallSuper
import androidx.lifecycle.ViewModel
import com.jdroid.android.utils.TagUtils
import com.jdroid.java.utils.LoggerUtils

abstract class AbstractViewModel : ViewModel() {

    @CallSuper
    override fun onCleared() {
        super.onCleared()
        LoggerUtils.getLogger(getTag()).info("Clearing view model")
    }

    protected fun getTag(): String {
        return TagUtils.getTag(javaClass)
    }
}
